src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1031,7 +1010,7 @@ where
     reclock_op.build(move |mut capabilities| {
         capabilities.clear();
 
-        let metrics_name = upstream_name.clone().unwrap_or_else(|| name.clone());
+        let metrics_name = name.clone();
         let mut source_metrics =
             SourceMetrics::new(&base_metrics, &metrics_name, id, &worker_id.to_string());
```

Comment:
nit: doesn't seem like it's used elsewhere so could just `&name` in the `SourceMetrics` constructor?

Reply:
Well spotted, thanks!